src/writer/proposal.ts

```diff
@@ -186,16 +193,33 @@ export async function verify(body): Promise<any> {
       space.id,
       body.address
     );
-    const [dayLimit, monthLimit] = getSpaceLimits(space);
 
-    if (dayCount >= dayLimit || monthCount >= monthLimit)
+    const spaceTypeWithEcosystem = await getSpaceType(space, true);
+    if (
+      dayCount >= (await getSpaceProposalsLimits(spaceTypeWithEcosystem, 'day')) ||
+      monthCount >= (await getSpaceProposalsLimits(spaceTypeWithEcosystem, 'month'))
+    )
       return Promise.reject('proposal limit reached');
-    if (!isAuthorized && activeProposalsByAuthor >= ACTIVE_PROPOSAL_BY_AUTHOR_LIMIT)
+    if (
+      !isAuthorized &&
+      activeProposalsByAuthor >= (await getLimit('space.active_proposal_limit_per_author'))
+    )
       return Promise.reject('active proposal limit reached for author');
   } catch (e) {
     capture(e);
     return Promise.reject('failed to check proposals limit');
   }
+
+  const spaceType = await getSpaceType(space);
+  const bodyLengthLimit = await getLimit(`space.${spaceType}.body_limit`);
+  if (msg.payload.body.length > bodyLengthLimit) {
+    return Promise.reject(`proposal body length can not exceed ${bodyLengthLimit} characters`);
+  }
+
+  const choicesLimit = await getLimit(`space.${spaceType}.choices_limit`);
```

Comment:
I think we should get all limits all at once per `writer` similar to UI and reuse them instead of getting one at a time for multiple times